pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -157,8 +157,8 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(env *model.Environme
 	// We'll try to return any listeners we successfully marshaled; if we have none, we'll emit the error we built up
 	err = errs.ErrorOrNil()
 	if len(listeners) == 0 {
-		log.Errorf("buildGatewayListeners: Have zero listeners: %v", err.Error())
-		return []*xdsapi.Listener{}, nil
+		log.Error("buildGatewayListeners: Have zero listeners")
+		return []*xdsapi.Listener{}, err
```

Comment:
I don't think we want to return `err` here, as an empty set of listeners is a valid value. Instead, lets just remove the call to `Error()` (which is unnecessary anyway):

```go
	log.Errorf("buildGatewayListeners: Have zero listeners: %v", err)
	return []*xdsapi.Listener{}, nil
```

Reply:
Yes,I agree with you.